### ISSUE TITLE
Issue with obtaining logits.

### ISSUE THREAD

**Issue by shuaiwang2022** [vllm-project/vllm#2220]
![image](https://github.com/vllm-project/vllm/assets/56470984/f150d087-a874-48a1-acbc-c7d24537d1f9)

I used the CodeLlama model with the given prompt, but when obtaining the logits, the shape is torch.Size([32016]). Shouldn't it be torch.Size([4, 32016])?
